### DISCUSSION TITLE
Making email optional

### DISCUSSION THREAD

**Discussion by Megubit** [JakyeRU/Larascord#30] [Ideas]
I was wondering if it would be possible to add a configuration to make the email optional?
If I understand correctly, the current minimum scope is `identify&email`, if I wish to only use `identify` it will fail since there is no email provided.

I assume it is possible to just remove the `if (empty($user->email))` check in `DiscordController.php`, but I was wondering if the email is mandatory for some security reason?

This might be just my own case, but I don't really want to store in the DB anything besides the user ID (ofc including the tokens, ect.), and for some smaller apps, I think it gives a sense of ease to the user when they see that the app only requires the ID and nothing else.

**Comment by ValentinNeaga** [comment]
Hello @Megubit ,

Thanks for bringing this issue to my attention. The e-mail was set to be mandatory because at the moment only the `identify&email` scope works.

Though, you've made a good point here. Many users would be skeptical about sharing their e-mail addresses, especially with smaller applications.

This has been solved in the version [@3.1.0](https://github.com/JakyeRU/Larascord/releases/tag/v3.1.0) by making it mandatory only if the `email` scope is used.

**Reply by Megubit** [reply]
Awesome, thank you for the new version 👍
